src/platform/forms-system/src/js/helpers.js

```diff
@@ -701,3 +701,20 @@ export function transformForSubmit(
 
   return JSON.stringify(withoutViewFields, replacer) || '{}';
 }
+
+/**
+ * @param {Object} uiSchema A nested `uiSchema`
+ * @returns {Object} The closest `uiSchema` object that holds the `uiSchema` keys
+ */
+export const checkForNestedUISchema = uiSchema => {
+  const uiSchemaKeys = Object.keys(uiSchema);
+  const isThisAUISchemaObj = !!uiSchemaKeys.find(uiSchemaKey =>
+    uiSchemaKey.includes('ui:'),
+  );
+  for (const uiSchemaKey of uiSchemaKeys) {
+    if (isThisAUISchemaObj) break;
+    const nestedUISchema = uiSchema[uiSchemaKey];
+    return checkForNestedUISchema(nestedUISchema);
+  }
+  return uiSchema;
```

Comment:
We assign the boolean `isThisAUISchemaObj` before we go into the loop, so why is the check done inside the loop with a break? If I inline the `nestedUISchema` value and pull the conditional to outside of the loop, this is what I get:

```js
  const isThisAUISchemaObj = !!uiSchemaKeys.find(uiSchemaKey =>
    uiSchemaKey.includes('ui:'),
  );
  
  // We only want to return the `checkForNestedUISchema` call if this is NOT a UISchema object
  if (!isThisAUISchemaObj) {
    for (const uiSchemaKey of uiSchemaKeys) {
      return checkForNestedUISchema(uiSchema[uiSchemaKey]);
    }
  }
  return uiSchema;
```

In your version as well as mine the loop will only run once, so why have it? Did you mean to use `continue` instead?

Reply:
I was intending to break out of the loop if the current object had the uiSchema properties. I like your approach as it seems more succinct.